src/routes/safe/components/Settings/ManageOwners/AddOwnerModal/index.tsx

```diff
@@ -25,12 +26,13 @@ export type OwnerValues = {
 export const sendAddOwner = async (
   values: OwnerValues,
   safeAddress: string,
-  safeVersion: string,
   txParameters: TxParameters,
   dispatch: Dispatch,
+  connectedWalletAddress: string,
 ): Promise<void> => {
-  const gnosisSafe = getGnosisSafeInstanceAt(safeAddress, safeVersion)
-  const txData = gnosisSafe.methods.addOwnerWithThreshold(values.ownerAddress, values.threshold).encodeABI()
+  const sdk = await getSafeSDK(connectedWalletAddress, safeAddress)
+  const safeTx = await sdk.getAddOwnerTx(values.ownerAddress, +values.threshold)
```

Comment:
You should check that it isn't `NaN`.

Reply:
I mean, it's better to be safe than sorry but if the SDK throws then I suppose you can leave it. I assume there is a catch block here?